lib/experimental/interactivity-api/scripts.php

```diff
@@ -18,7 +18,7 @@ function gutenberg_interactivity_move_interactive_scripts_to_the_footer() {
 	// Move all the view scripts of the interactive blocks to the footer.
 	$registered_blocks = \WP_Block_Type_Registry::get_instance()->get_all_registered();
 	foreach ( array_values( $registered_blocks ) as $block ) {
-		if ( isset( $block->supports['interactivity'] ) && $block->supports['interactivity'] ) {
+		if ( isset( $block->supports['interactivity'] ) && $block->supports['interactivity'] && gutenberg_should_block_use_interactivity_api( $block->name ) ) {
```

Comment:
Good catch. But what about removing `supports.interactivity` from the `block.json` and adding it dynamically instead? We already have a place for that:

```php
function gutenberg_block_core_navigation_update_interactive_metadata( $metadata ) {
  if ( 'core/navigation' === $metadata['name'] ) {
    $metadata['viewScript']                = array( 'file:./view-interactivity.min.js' );
    $metadata['supports']['interactivity'] = true;
  }
  return $metadata;
}
add_filter( 'block_type_metadata', 'gutenberg_block_core_navigation_update_interactive_metadata', 10, 1 );
```

I think it'll be safer because if we add more functionality based on the `interactivity` property in the future, the old implementation could break.

Reply:
Like d33578c64c4?

Reply:
Exactly, yes. Thank you.

_By the way, I'd like to replace this part with the new `defer` loading strategy as soon as the minimum required version of Gutenberg is WP 6.3._